scripts/setup_osx.sh

```diff
@@ -1,38 +1,248 @@
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: MuJoCo couldn't be set up\e[0m"
-  exit 1
+parse_commandline ()
+{
+  while test $# -gt 0
+  do
+    _key="$1"
+    case "$_key" in
+      --no-set-envvar|--set-envvar)
+        _arg_set_envvar="on"
+        test "${1:0:5}" = "--no-" && _arg_set_envvar="off"
+        ;;
+      -h|--help)
+        print_help
+        exit 0
+        ;;
+      -h*)
+        print_help
+        exit 0
+        ;;
+      *)
+        _positionals+=("$1")
+        ;;
+    esac
+    shift
+  done
+}
+
+
+handle_passed_args_count ()
+{
+  _required_args_string="'mjkey-path'"
+  error_msg="$(echo "FATAL ERROR: Not enough positional arguments - we" \
+    "require exactly 1 (namely: $_required_args_string), but got only" \
+    "${#_positionals[@]}.")"
+  test ${#_positionals[@]} -ge 1 || _PRINT_HELP=yes die "${error_msg}" 1
+  error_msg="$(echo "FATAL ERROR: There were spurious positional arguments" \
+    "--- we expect exactly 1 (namely: $_required_args_string), but got" \
+    "${#_positionals[@]} (the last one was: '${_positionals[*]: -1}').")"
+  test ${#_positionals[@]} -le 1 || _PRINT_HELP=yes die "${error_msg}" 1
+}
+
+assign_positional_args ()
+{
+  _positional_names=('_arg_mjkey_path' )
+
+  for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+  do
+    error_msg="$(echo "Error during argument parsing, possibly an Argbash" \
+      "bug.")"
+    eval "${_positional_names[ii]}=\${_positionals[ii]}" || \
+      die "${error_msg}" 1
+  done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Utility functions
+print_error() {
+  printf "\e[0;31m%s\e[0m" "${1}"
+}
+
+print_warning() {
+  printf "\e[0;33m%s\e[0m" "${1}"
+}
+
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey_path}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey_path} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(readlink -f $(dirname $0)/..)"
+cd "${GARAGE_DIR}"
+
+# File where environment variables are stored
+BASH_PROF="${HOME}/.bash_profile"
+
+# Install dependencies
+echo "Installing garage dependencies"
+echo "You will probably be asked for your sudo password"
+
+# Homebrew is required first to install the other dependencies
+hash brew 2>/dev/null || {
+  # Install the Xcode Command Line Tools
+  xcode-select --install
+  # Install Homebrew
+  /usr/bin/ruby -e "$(curl -fsSL \
+    https://raw.githubusercontent.com/Homebrew/install/master/install)"
+}
+
+# For installing garage: wget, bzip2, unzip, git
+# For building glfw: cmake
+# Required for OpenAI gym: cmake boost boost-python ffmpeg sdl2 swig wget
+# Required for OpenAI baselines: cmake openmpi
+brew update
+brew install \
+  wget \
+  bzip2 \
+  unzip \
+  git \
+  cmake \
+  boost \
+  boost-python \
+  ffmpeg \
+  sdl2 \
+  swig \
+  openmpi
+
+# Build GLFW because the brew version is too old
+GLFW_DIR="$(mktemp -d)/glfw"
```

Comment:
Is it true that the brew version is too old? Does this even work?

Reply:
I'm actually using the brew version of glfw right now and it works fine for me.

Reply:
Ok, then I will remove the compilation of glfw and add it to the brew.